src/pyodide.js

```diff
@@ -679,11 +679,43 @@ def temp(Module):
   globals = __main__.__dict__
   globals.update(builtins.__dict__)
 
+  global saved_globals
+  saved_globals = {}
+  saved_globals.update(globals)
+
   Module.version = pyodide.__version__
   Module.globals = globals
   Module.builtins = builtins.__dict__
   Module.pyodide_py = pyodide
 `);
+  Module.resetState = function() {
+    pyodide.globals.clear();
+    Module.runPythonSimple(`
```

Comment:
I think we need to add documentation on runPythonSimple and when it should be used instead of runPython. Right now that's not clear to me.

Reply:
The primary point of `runPythonSimple` is as a bootstrap method. It doesn't depend on any of the other code in `core` so it will work as soon as Python is initialized even if everything else goes really wrong.

The differences are: 
1. `runPythonSimple` doesn't return anything (and so won't leak PyProxies)
2. `runPythonSimple` doesn't require access to any state on the `pyodide_js` module. 
3. `runPython` uses `pyodide.eval_code`, whereas `runPythonSimple` uses `PyRun_String` which is the C API for `eval` / `exec`.
4. `runPythonSimple` runs with `globals` a separate dict which I've called `init_dict`
5. `runPythonSimple` doesn't dedent the argument

Reply:
It would be good to add that to the docstring of `runPythonSimple` (not in this PR)